src/java/voldemort/utils/ExceptionUtils.java

```diff
@@ -0,0 +1,15 @@
+package voldemort.utils;
+
+/**
+ * Static utility functions to deal with exceptions.
+ */
+public class ExceptionUtils {
+    public static boolean recursiveClassEquals(Throwable throwableToInspect, Class throwableToLookFor) {
```

Comment:
@arunthirupathi, check this out.

I think this is the type of stuff we would have needed when you were dealing with the client-side exception getting wrapped once too many times.

It seems to me like it's rarely a good idea to look at the cause of an exception, but if you're gonna do it, it usually makes sense to recurse all the way into the call stack, rather than expect you'll always find what you need exactly n levels deep.

If you agree, can we change the error handling code you had worked on to use this new utility function? This way, we'd get un-modified stacktraces in our logs, and make the code a bit cleaner.

-F